tsdb/index/tsi1/index_test.go

```diff
@@ -411,3 +579,271 @@ func (idx *Index) CreateSeriesSliceIfNotExists(a []Series) error {
 	}
 	return idx.CreateSeriesListIfNotExists(keys, names, tags)
 }
+
+var tsiditr tsdb.SeriesIDIterator
+
+// Calling TagValueSeriesIDIterator on the index involves merging several
+// SeriesIDSets together.BenchmarkIndex_TagValueSeriesIDIterator, which can have
+// a non trivial cost. In the case of `tsi` files, the mmapd sets are merged
+// together. In the case of tsl files the sets need to are cloned and then merged.
+//
+// Typical results on an i7 laptop
+// BenchmarkIndex_IndexFile_TagValueSeriesIDIterator/78888_series_TagValueSeriesIDIterator/cache-8   	 2000000	       643 ns/op	     744 B/op	      13 allocs/op
+// BenchmarkIndex_IndexFile_TagValueSeriesIDIterator/78888_series_TagValueSeriesIDIterator/no_cache-8      10000	    130749 ns/op	  124952 B/op	     350 allocs/op
+func BenchmarkIndex_IndexFile_TagValueSeriesIDIterator(b *testing.B) {
+	var err error
+	sfile := NewSeriesFile()
+	// Load index
+	idx := tsi1.NewIndex(sfile.SeriesFile, "foo",
+		tsi1.WithPath("testdata/index-file-index"),
+		tsi1.DisableCompactions(),
+	)
+	defer sfile.Close()
+
+	if err = idx.Open(); err != nil {
+		b.Fatal(err)
+	}
+	defer idx.Close()
+
+	runBenchMark := func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			tsiditr, err = idx.TagValueSeriesIDIterator([]byte("m4"), []byte("tag0"), []byte("value4"))
+			if err != nil {
+				b.Fatal(err)
+			} else if tsiditr == nil {
+				b.Fatal("got nil iterator")
+			}
+		}
+	}
+
+	// This benchmark will merge eight bitsets each containing ~10,000 series IDs.
+	b.Run("78888 series TagValueSeriesIDIterator", func(b *testing.B) {
+		b.ReportAllocs()
+		b.Run("cache", func(b *testing.B) {
+			runBenchMark(b)
```

Comment:
this should probably set `tsi1.EnableBitsetCache = true` to avoid ordering issues with other possible benchmarks? and/or default changes.